erigon-lib/state/aggregator_v3.go

```diff
@@ -774,16 +750,18 @@ func (ac *AggregatorV3Context) CanUnwindBeforeBlockNum(blockNum uint64, tx kv.Tx
 	return blockNumWithCommitment, true, nil
 }
 
-// returns true if we can prune something already aggregated
-func (ac *AggregatorV3Context) nothingToPrune(tx kv.Tx, untilTxNum uint64) bool {
-	return dbg.NoPrune() || (!ac.account.CanPruneUntil(tx, untilTxNum) &&
-		!ac.storage.CanPruneUntil(tx, untilTxNum) &&
-		!ac.code.CanPruneUntil(tx, untilTxNum) &&
-		!ac.commitment.CanPruneUntil(tx, untilTxNum) &&
-		!ac.logAddrs.CanPrune(tx) &&
-		!ac.logTopics.CanPrune(tx) &&
-		!ac.tracesFrom.CanPrune(tx) &&
-		!ac.tracesTo.CanPrune(tx))
+func (ac *AggregatorV3Context) somethingToPrune(tx kv.Tx, untilTxNum uint64) bool {
```

Comment:
do we need `untilTxNum` param or better always watch latest snapshot?

Reply:
it comes from `ac.somethingToPrune(tx, ac.maxTxNumInDomainFiles(false))` and main thing in this line is: look at `Domain only`, not on history/idx, not on combination of 3. Because: there may be no history (commitment/pruned_node) 

I guess: we can call `ac.maxTxNumInDomainFiles(false)` inside `somethingToPrune`.